File: app/src/main/java/com/arad/care4pets/AddPetActivity.java
package com.arad.care4pets;

import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;

public class AddPetActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_add_pet);

        EditText etName = findViewById(R.id.etPetName);
        EditText etSpecies = findViewById(R.id.etPetSpecies);
        EditText etAge = findViewById(R.id.etPetAge);
        EditText etNotes = findViewById(R.id.etPetNotes);
        Button btnSave = findViewById(R.id.btnSavePet);

        btnSave.setOnClickListener(v -> {
        String name = etName.getText().toString().trim();
        String species = etSpecies.getText().toString().trim();
        String ageText = etAge.getText().toString().trim();
        String notes = etNotes.getText().toString().trim();

        if (name.isEmpty() || species.isEmpty() || ageText.isEmpty()) {
            Toast.makeText(this, "Please fill required fields", Toast.LENGTH_SHORT).show();
            return;
        }

        int newId;
        if (!FakeRepository.pets.isEmpty()) {
            int maxId = 0;
            for (Pet pet : FakeRepository.pets) {
                if (pet.getId() > maxId) {
                    maxId = pet.getId();
                }
            }
            newId = maxId + 1;
        } else {
            newId = 1;
        }

        Pet pet = new Pet(newId, name, species, Integer.parseInt(ageText), notes);
        FakeRepository.pets.add(pet);

        Toast.makeText(this, "Pet added", Toast.LENGTH_SHORT).show();
        finish();
    });
    }
}

File: app/src/main/java/com/arad/care4pets/AddPetActivity.kt
package com.arad.care4pets

import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class AddPetActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_add_pet)

        val etName = findViewById<EditText>(R.id.etPetName)
        val etSpecies = findViewById<EditText>(R.id.etPetSpecies)
        val etAge = findViewById<EditText>(R.id.etPetAge)
        val etNotes = findViewById<EditText>(R.id.etPetNotes)
        val btnSave = findViewById<Button>(R.id.btnSavePet)

        btnSave.setOnClickListener {
            val name = etName.text.toString().trim()
            val species = etSpecies.text.toString().trim()
            val ageText = etAge.text.toString().trim()
            val notes = etNotes.text.toString().trim()

            if (name.isEmpty() || species.isEmpty() || ageText.isEmpty()) {
                Toast.makeText(this, "Please fill required fields", Toast.LENGTH_SHORT).show()
                return@setOnClickListener
            }

            val newId = (FakeRepository.pets.maxOfOrNull { it.id } ?: 0) + 1
            val pet = Pet(newId, name, species, ageText.toInt(), notes)
            FakeRepository.pets.add(pet)

            Toast.makeText(this, "Pet added", Toast.LENGTH_SHORT).show()
            finish()
        }
    }
}
